src/neo4j_graphrag/tool.py

```diff
@@ -0,0 +1,224 @@
+from typing import Any, Dict, List, Callable, Optional
+
+
+class ParameterType(str, Enum):
+    """Enum for parameter types supported in tool parameters."""
+
+    STRING = "string"
+    INTEGER = "integer"
+    NUMBER = "number"
+    BOOLEAN = "boolean"
+    OBJECT = "object"
+    ARRAY = "array"
+
+
+class ToolParameter:
+    """Base class for all tool parameters."""
+
+    def __init__(self, description: str, required: bool = False):
+        self.description = description
+        self.required = required
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Convert the parameter to a dictionary format."""
+        raise NotImplementedError("Subclasses must implement to_dict")
+
+
+class StringParameter(ToolParameter):
+    """String parameter for tools."""
+
+    def __init__(
+        self, description: str, required: bool = False, enum: Optional[List[str]] = None
+    ):
+        super().__init__(description, required)
+        self.enum = enum
+
+    def to_dict(self) -> Dict[str, Any]:
+        result: Dict[str, Any] = {
+            "type": ParameterType.STRING,
+            "description": self.description,
+        }
+        if self.enum:
+            result["enum"] = self.enum
+        return result
+
+
+class IntegerParameter(ToolParameter):
+    """Integer parameter for tools."""
+
+    def __init__(
+        self,
+        description: str,
+        required: bool = False,
+        minimum: Optional[int] = None,
+        maximum: Optional[int] = None,
+    ):
+        super().__init__(description, required)
+        self.minimum = minimum
+        self.maximum = maximum
+
+    def to_dict(self) -> Dict[str, Any]:
+        result: Dict[str, Any] = {
+            "type": ParameterType.INTEGER,
+            "description": self.description,
+        }
+        if self.minimum is not None:
+            result["minimum"] = self.minimum
+        if self.maximum is not None:
+            result["maximum"] = self.maximum
+        return result
+
+
+class NumberParameter(ToolParameter):
+    """Number parameter for tools."""
+
+    def __init__(
+        self,
+        description: str,
+        required: bool = False,
+        minimum: Optional[float] = None,
+        maximum: Optional[float] = None,
+    ):
+        super().__init__(description, required)
+        self.minimum = minimum
+        self.maximum = maximum
+
+    def to_dict(self) -> Dict[str, Any]:
+        result: Dict[str, Any] = {
+            "type": ParameterType.NUMBER,
+            "description": self.description,
+        }
+        if self.minimum is not None:
+            result["minimum"] = self.minimum
+        if self.maximum is not None:
+            result["maximum"] = self.maximum
+        return result
+
+
+class BooleanParameter(ToolParameter):
+    """Boolean parameter for tools."""
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"type": ParameterType.BOOLEAN, "description": self.description}
+
+
+class ObjectParameter(ToolParameter):
```

Comment:
Nice! It's maybe a bit verbose, would be great if users could provide tool definition as dict, wdyt? I think converting the `*Parameter` to Pydantic model would make this easily possible (and we could also get rid of the `to_dict` methods).

Reply:
Great idea!

Reply:
Pushed this change now